test/testunusedvar.cpp

```diff
@@ -4791,10 +4791,54 @@ class TestUnusedVar : public TestFixture {
             "public:\n"
             "   uint32_t u{1};\n"
             "};\n"
-            "int main(int argc, char argv[]) {\n"
+            "void f() {\n"
             "   E e;\n"
             "}");
         ASSERT_EQUALS("[test.cpp:7]: (style) Unused variable: e\n", errout.str());
+
+        // non-empty constructor
+        functionVariableUsage(
+            "class F {\n"
+            "public:\n"
+            "   F() {\n"
+            "       int i =0;\n"
+            "       (void) i;"
+            "   }\n"
+            "};\n"
+            "void f() {\n"
+            "   F f;\n"
+            "}");
+        ASSERT_EQUALS("", errout.str());
```

Comment:
This might actually be something worth detecting as unused - nothing for this PR though.

Maybe this should even become a separate check to detect that the function does nothing. There's a whole chain. I wonder how to handle this...

function without effect -> empty c'tor -> no side-effects -> unused variable